libcomposefs/lcfs-mount.c

```diff
@@ -680,7 +711,21 @@ int lcfs_mount_fd(int fd, const char *mountpoint, struct lcfs_mount_options_s *o
 		return -1;
 	}
 
+	int memfd = -1;
+	if (options->flags & LCFS_MOUNT_FLAGS_COPY) {
+		memfd = lcfs_mount_copy_and_seal(fd);
+		if (memfd < 0) {
+			return memfd;
+		} else {
```

Comment:
I'd rather skip the else clause in this kind of "didn't get error" codepath. I.e.:
```
if (memfd < 0) {
  return memfd:
}
state.fd = memfd;
```